app/feature/feature-claim-details/src/main/kotlin/com/hedvig/android/feature/claim/details/data/UploadFileUseCase.kt

```diff
@@ -0,0 +1,87 @@
+package com.hedvig.android.feature.claim.details.data
+
+import android.content.ContentResolver
+import android.net.Uri
+import arrow.core.Either
+import arrow.core.raise.either
+import arrow.core.raise.ensureNotNull
+import arrow.retrofit.adapter.either.networkhandling.CallError
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.core.common.android.FileService
+import com.hedvig.android.core.retrofit.toErrorMessage
+import com.hedvig.android.logger.LogPriority
+import com.hedvig.android.logger.logcat
+import kotlinx.serialization.Serializable
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okio.BufferedSink
+import okio.source
+import retrofit2.http.Multipart
+import retrofit2.http.POST
+import retrofit2.http.Part
+import retrofit2.http.Query
+
+internal interface UploadFileUseCase {
+  suspend fun invoke(url: String, uri: Uri): Either<ErrorMessage, UploadSuccess>
+}
+
+internal data class UploadFileUseCaseImpl(
+  private val uploadFileService: UploadFileService,
+  private val fileService: FileService,
+  private val contentResolver: ContentResolver,
+) : UploadFileUseCase {
+  override suspend fun invoke(url: String, uri: Uri): Either<ErrorMessage, UploadSuccess> = either {
+    val claimId = Uri.parse(url).getQueryParameter("claimId") ?: raise(ErrorMessage("No claim id found in url"))
```

Comment:
You are right, this is currently the same as the claim id. I will keep it like this though since this URL is set on the claim, so it may change in the future.